matrix_content_scanner/config.py

```diff
@@ -181,3 +206,6 @@ def __init__(self, config_dict: Dict[str, Any]):
         self.crypto = CryptoConfig(**(config_dict.get("crypto") or {}))
         self.download = DownloadConfig(**(config_dict.get("download") or {}))
         self.result_cache = ResultCacheConfig(**(config_dict.get("result_cache") or {}))
+
+        # Check that we can read the pickle file, or create it if it doesn't exist.
+        _check_file_path(self.crypto.pickle_path)
```

Comment:
`_check_file_path` doesn't appear to create the file, in contrast to what the comment says.

In general: I don't think that separating the checks from the actual use of the file is good, for a number of reasons:
 * it tightly couples this config code to the way that the file is used. (For example: we have embedded the knowledge here that we only need read access to the file if it already exists. That's fine as far as it goes, but is ripe for getting out of sync if we change the way `CryptoHandler` works.)
 * You're kinda relying on `os.access` and `open` to interact in the way you expect. One example is that `os.access` checks with the *real* userid, whereas `open` uses the *effective* userid. Admittedly they are unlikely to be different, but it feels like an unnecessary assumption, and ties us quite hard to unix-like filesystems. Another example: the disk might be full (so `os.access` is happy, but you still can't write the file).
 * Just because the file exists/is writable here, doesn't necessarily mean that it will still be so when we come to use it later. For example, the file might get deleted in the meantime, or created with different permissions.

All in all: I think it would be better to make sure we construct `CryptoHandler` at startup, and handle the exceptions from `open` in a way that gives us nicer output.

Reply:
> `_check_file_path` doesn't appear to create the file, in contrast to what the comment says.

Ah I phrased this poorly, it meant to read as "that we can [...] create it if it doesn't exist".

> I don't think that separating the checks from the actual use of the file is good, for a number of reasons:

Yeah good points. I'll try to rework that differently.

Reply:
> I think it would be better to make sure we construct CryptoHandler at startup, and handle the exceptions from open in a way that gives us nicer output.

It's worth noting that `CryptoHandler` _will_ be constructed at startup: it will be constructed by one of the `/..._encrypted` http resources (such as [this one](https://github.com/matrix-org/matrix-content-scanner-python-wip/blob/88ab968106537a1451e9b11771a5cd24df0b2376/matrix_content_scanner/servlets/download.py#L54)), which are constructed by the [`HTTPServer` constructor](https://github.com/matrix-org/matrix-content-scanner-python-wip/blob/main/matrix_content_scanner/httpserver.py#L34-L63). An `HTTPServer` is constructed on `MatrixContentScanner.start`.